src/server/conn/http1.rs

```diff
@@ -116,25 +111,20 @@ where
     /// # Panics
     /// This method will panic if this connection is using an h2 protocol.
     pub fn into_parts(self) -> Parts<I, S> {
-        self.try_into_parts()
-            .unwrap_or_else(|| panic!("h2 cannot into_inner"))
+        let (io, read_buf, dispatch) = self.conn.into_inner();
+        Parts {
+            io,
+            read_buf,
+            service: dispatch.into_service(),
+            _inner: (),
+        }
     }
 
     /// Return the inner IO object, and additional information, if available.
     ///
     ///
-    /// TODO:(mike) does this need to return none for h1 or is it expected to always be present? previously used an "unwrap"
-    /// This method will return a `None` if this connection is using an h2 protocol.
     pub fn try_into_parts(self) -> Option<Parts<I, S>> {
```

Comment:
In fact, this function now seems unnecessary in its entirety, ya?

Reply:
It is correct.  I considered that and then left it in since removing it would be a breaking change... I'll remove 😸